virtualizarr/manifests/manifest.py

```diff
@@ -54,6 +65,110 @@ def with_validation(
         return ChunkEntry(path=path, offset=offset, length=length)
 
 
+class ManifestStore(Store):
+    supports_writes: bool = False
+    supports_deletes: bool = False
+    supports_partial_writes: bool = False
+    supports_listing: bool = True
+
+    fs: AsyncFileSystem
+    vds: xr.Dataset
+
+    def __init__(self, fs, vds, read_only=True):
+        super().__init__(read_only=read_only)
+        self.fs = fs
+        self.vds = vds
+
+    async def clear(self) -> None:
+        self.fs = None
+        self.vds = None
+
+    def __str__(self) -> str:
+        return f"manifest://{id(self.vds)}"
```

Comment:
We might want to build a custom repr using the `xarray.Dataset` repr here. Low priority suggestion though.